Geometry_Engine/Compute/OutlinesFromLines.cs

```diff
@@ -0,0 +1,427 @@
+        {
+            // Transform the lines to global XY plane
+            TransformMatrix toGlobal = lines.TransformToGlobalXY();
+            if (toGlobal == null)
+            {
+                BH.Engine.Base.Compute.RecordWarning("Some of the lines have been ignored in the process of creating outlines because they were all collinear.");
+                return null;
+            }
+
+            List<Line> transformed = lines.Select(x => x.Transform(toGlobal)).ToList();
+
+            // Turn the lines into a graph represented as a dictionary of nodes and adjoining edges
+            Dictionary<Point, List<Line>> graph = new Dictionary<Point, List<Line>>();
+            foreach (Line line in transformed)
+            {
+                if (!graph.ContainsKey(line.Start))
+                    graph.Add(line.Start, new List<Line>());
+
+                if (!graph.ContainsKey(line.End))
+                    graph.Add(line.End, new List<Line>());
+
+                graph[line.Start].Add(line);
+                graph[line.End].Add(line);
+            }
+
+            // Start at leftmost node
+            Point leftmost = graph.Keys.First();
+            foreach (Point node in graph.Keys)
+            {
+                if (node.X < leftmost.X)
+                    leftmost = node;
+            }
+
+            // Find edge starting at start node with leftmost dir
+            Line currentEdge = null;
+            Vector currentDir = null;
+            foreach (Line l in graph[leftmost])
+            {
+                var dir = l.Direction();
+                if (l.Start != leftmost)
+                    dir = dir.Reverse();
+
+                if (currentDir == null || currentDir.Y < dir.Y)
+                {
+                    currentEdge = l;
+                    currentDir = dir;
+                }
+            }
+
+            List<Line> outline = new List<Line> { currentEdge };
+            Point currentNode = leftmost;
+
+            // Look for subsequent edges with leftmost dirs compared to the dir of current edge
+            while (true)
+            {
+                if (currentNode == currentEdge.Start)
+                    currentNode = currentEdge.End;
+                else
+                    currentNode = currentEdge.Start;
+
+                if (currentNode == leftmost)
+                    break;
+
+                List<Vector> nodeDirs = new List<Vector>();
+                foreach (Line l in graph[currentNode])
+                {
+                    Vector dir = l.Direction();
+                    if (l.Start != currentNode)
+                        dir = dir.Reverse();
+
+                    nodeDirs.Add(dir);
+                }
+
+                Vector newDir = null;
+                Line newEdge = null;
+
+                // To avoid computing heavy signed angle, first check only dirs to the left from the current dir
```

Comment:
We are looking for directions to the left from the current direction, i.e. for edges in the left part of the space cut with infinite line defined by current node and current dir.

On an example, let's assume the green rectangle is our space (effectively bounds of points we process), blue is current node and dir. When we position ourselves at the node and look in the direction of dir, we first check for edges in orange area that meet at the node. Only if there is none of them, we check the white area. The reason for it, as in the code comment, is to avoid heave signed angle computations.

![image](https://github.com/BHoM/BHoM_Engine/assets/26874773/76c47480-5804-4a4f-a4a9-639051299615)